### PR TITLE
bump bazel build to go1.8.1 and remove invalid unit tests

### PR DESCRIPTION
**Pull request by mikedanese**
part of https://github.com/kubernetes/kubernetes/issues/38228

I firmly believe that unit tests that check error strings are incorrect unit tests. If we care about what type of error is returned, we need to use public error types. Anywhere we are using generic errors, we don't care other then that we saw an error.

### CODE REVIEW

**Comment by mikedanese** [comment]
@thockin for top level approval

**Comment by ixdy** [comment]
is it worth refactoring the unit tests to add back in the functionality we're removing (verifying that a particular kind of error happened)?

**Comment by ixdy** [comment]
x-ref #41636

cc @luxas

**Comment by mikedanese** [comment]
I firmly believe that unit tests that check error strings are incorrect unit tests. If we care about what type of error is returned, we need to use public error types. Anywhere we are using generic errors, we don't care other then that we saw an error.

**Comment by mikedanese** [comment]
@k8s-bot cvm gce e2e test this

**Comment by spxtr** [comment]
This seems fine, but it puts us in a bit of a strange in-between state where we run unit tests on 1.8 in `pull-kubernetes-bazel` and 1.7 in `pull-kubernetes-unit`.

(can you amend the last comment to not say "lot's"...)

File: staging/src/k8s.io/apimachinery/pkg/runtime/embedded_test.go [hunk 1]
```diff
@@ -240,8 +239,11 @@ func TestNestedObject(t *testing.T) {
 	// the external representation
 	var decodedViaJSON EmbeddedTest
 	err = json.Unmarshal(wire, &decodedViaJSON)
-	if err == nil || !strings.Contains(err.Error(), "unmarshal object into Go value of type runtime.Object") {
-		t.Fatalf("Unexpected decode error %v", err)
+	if err == nil {
+		t.Fatalf("Expeceted decode error")
```
Comment (spxtr) [inline staging/src/k8s.io/apimachinery/pkg/runtime/embedded_test.go]: `t.Fatal("Expected decode error")`

**Comment by spxtr** [comment]
LGTM other than nit.

**Comment by mikedanese** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by luxas** [comment]
ping @thockin for top-level approval

/approve

(although I think that @ixdy's, @spxtr's, @mikedanese's and my approval should be sufficent for this specific PR though)

**Comment by ixdy** [comment]
lgtm. still need approval...

**Comment by thockin** [comment]
Neither the title nor the first comment on this PR explains why it has test changes...

**Comment by ixdy** [comment]
re: gofmt failure - I wonder if we need to update gofmt for 1.8?

**Comment by luxas** [comment]
@ixdy maybe -- but doesn't it run inside of the build image env?

**Comment by ixdy** [comment]
@mikedanese is out today, so I've attempted to update this PR for him. I rebased it against latest master and run `gofmt -s -w` against `pkg/proxy/iptables/proxier_test.go`.

It looks like gofmt 1.7.5 and 1.8.1 are happy with this file reformatted. We'll see if golint in CI complains.

**Comment by ixdy** [comment]
@thockin I updated the PR title and copied @mikedanese's explanation for removal of unit tests to the first comment. thanks!

**Comment by luxas** [comment]
Perfect @ixdy, thanks for picking this up!

**Pull request merged**